controller/user.js

```diff
@@ -0,0 +1,237 @@
+      }),
+    );
+  } catch (error) {
+    console.log(error);
+    next();
+  }
+};
+
+const logIn = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  try {
+    const validationByJoi = schema.validate({
+      email,
+      password,
+    });
+
+    if (validationByJoi.error) {
+      const validationErrorMessage = validationByJoi.error.details[0].message;
+      return res
+        .status(400)
+        .json(
+          genereteJSON(
+            "error",
+            400,
+            "error message",
+            validationErrorMessage || "Unsuccessful validation",
+          ),
+        );
+    }
+
+    const user = await service.getUserByEmail(email);
+
+    const result = await verifyPassword(password, user.password);
+
+    if (!result) {
+      return res
+        .status(401)
+        .json(
+          genereteJSON(
+            "error",
+            401,
+            "error message",
+            "Email or password is wrong",
+          ),
+        );
+    }
+
+    const payload = {
+      _id: user._id,
+    };
+
+    const token = jwt.sign(payload, SECRET, { expiresIn: 60 * 5 * 1000000 });
+
+    const responseBody = {
+      token,
+      user: { email: user.email, subscription: user.subscription },
+    };
+
+    res.json(genereteJSON("success", 200, "body", responseBody));
+  } catch (error) {
+    console.log(error);
+    next();
+  }
+};
+
+const logOut = async (req, res, next) => {
+  const { _id: id } = req.user;
+
+  try {
+    const user = await service.getUserById(id);
+    console.log(user);
+
+    if (!user) {
+      return res
+        .status(401)
+        .json(genereteJSON("error", 401, "error message", "Not authorized"));
+    }
+
+    if (user.token) {
+      await service.updateUserById(id, { token: null });
+    }
+
+    res.status(204).json({ mess: "test" });
+  } catch (error) {
+    console.log(error);
+    next();
+  }
+};
+
+const get = async (req, res, next) => {
+  const { _id: id } = req.user;
+
+  try {
+    const user = await service.getUserById(id);
+
+    if (!user) {
+      return res
+        .status(401)
+        .json(genereteJSON("error", 401, "error message", "Not authorized"));
+    }
+
+    res.json(genereteJSON("success", 200, "user", user));
+  } catch (error) {
+    console.log(error);
+    next();
+  }
+};
+
+const update = async (req, res, next) => {
+  const { _id: id } = req.user;
+  const { subscription } = req.body;
+  console.log(subscription);
+  let subscriptionType;
+
+  try {
+    const user = await service.getUserById(id);
+
+    if (!user) {
+      return res
+        .status(401)
+        .json(genereteJSON("error", 401, "error message", "Not authorized"));
+    }
+
+    switch (subscription) {
+      case "starter":
+        subscriptionType = "starter";
+        break;
+      case "pro":
+        subscriptionType = "pro";
+        break;
+      case "business":
+        subscriptionType = "business";
+        break;
+      default:
```

Comment:
Jeśli sprawdzisz w joi co przyszło tego nie będziesz potrzebować.

Reply:
Czytelniejsze będzie jeśli sprawdzisz w joi i jeśli wposzło dobrze to po prostu przypisz
subscriptionType = subscription